Add docs for the Mime component

fabpot opened the pull request:
docs for symfony/symfony#29896

<!--

If your pull request fixes a BUG, use the oldest maintained branch that contains
the bug (see https://symfony.com/roadmap for the list of maintained branches).

If your pull request documents a NEW FEATURE, use the same Symfony branch where
the feature was introduced (and `master` for features of unreleased versions).

-->

fabpot committed:
added docs for the Mime component

pborreli commented on components/mime.rst:
```diff
@@ -0,0 +1,63 @@
+    // $mimeTypes contains an array of mime types: ['application/mbox']
+
+    // guess a mime type for a file
+    $mimeType = $mimeTypes->guessMimeType('/some/path/to/image.gif);
```
miss the ending quote after `gif`

fabpot replied:
fixed

Pierstoval commented on reference/dic_tags.rst:
```diff
@@ -450,6 +451,15 @@ To add a new rendering strategy - in addition to the core strategies like
+**Purpose**: Add a custom mime type guesser to MimeTypes
+
+To add a custom mime type guesser - in addition to the core guessers - create a
+class that implements :class:`Symfony\\Component\\MimeType\\MimeTypes`, register
```
```suggestion
class that implements :class:`Symfony\\Component\\Mime\\MimeTypeGuesserInterface`, register
```

pborreli replied:
same line 24

fabpot replied:
fixed

Pierstoval commented on components/mime.rst:
```diff
@@ -0,0 +1,63 @@
+    {
+        public function isGuesserSupported(): bool
+        {
+            // return true when the guesser is supported (might depend on the OS for instance)
```
What about adding this?
```suggestion
            // Return true when the guesser is supported (might depend on the OS, dependencies or PHP extensions for instance)
```

Pierstoval commented on components/mime.rst:
```diff
@@ -0,0 +1,63 @@
+
+        public function guessMimeType(string $path): ?string
+        {
+            // return a mime type based on the content of the file stored in $path
```
```suggestion
            // Return a mime type based on the content of the file stored in $path.
```

Pierstoval commented on components/mime.rst:
```diff
@@ -0,0 +1,63 @@
+        public function guessMimeType(string $path): ?string
+        {
+            // return a mime type based on the content of the file stored in $path
+            // you MUST not use the filename to determine the mime type.
```
```suggestion
            // You MUST not use the filename to determine the mime type.
```

fabpot replied:
I like it this way

Pierstoval commented on components/mime.rst:
```diff
@@ -0,0 +1,63 @@
+Mime Types
+----------
+
+The :class:`Symfony\\Component\\MimeType\\MimeTypes` class manipulates the relationships
```
```suggestion
The :class:`Symfony\\Component\\Mime\\MimeTypes` class manipulates the relationships
```

fabpot replied:
fixed

javiereguiluz committed:
Proposed some rewords

Instead of adding comments about possible changes, I've decided to commit them. Revert totally or partially if you disagree with them. Thanks!

tony-tran commented on components/mime.rst:
```diff
@@ -0,0 +1,88 @@
+Adding a MIME Type Guesser
+~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+You can register your own MIME type guesser by creating a class that extends
```
Maybe you can change `extends` by `implements` ? WDYT ?

tony-tran commented on reference/dic_tags.rst:
```diff
@@ -450,6 +451,15 @@ To add a new rendering strategy - in addition to the core strategies like
+**Purpose**: Add a custom MIME type guesser to MimeTypes
+
+To add a custom MIME type guesser - in addition to the core guessers - create a
+class that implements :class:`Symfony\\Component\\Mime\\MimeTypes`, register it
```
Is it `MimeTypesInterface` ?

thewilkybarkid commented on components/mime.rst:
```diff
@@ -0,0 +1,88 @@
+==================
+
+    The MIME component provides utilities related to MIME types, such as
+    guessing the MIME type of a given file.
```
If the scope of the component is just media types rather than MIME as a whole, could it be named as such instead?

OskarStark commented:
Looks like the code was merged, can we merge this too and announce this new component?

javiereguiluz commented:
Closing because we've added all these contents to #11280, which documents the entire Mime component.
